### DISCUSSION TITLE
Code Coverage with Code Climate

### DISCUSSION THREAD

**Discussion by ksylvest** [community/community#26027] [Actions]
<p>I’m attempting to send code coverage to code climate using a github action:</p>
<p><a href="https://github.com/ksylvest/tights/blob/master/.github/workflows/test.yml#L17-L25" rel="nofollow noopener">https://github.com/ksylvest/tights/blob/master/.github/workflows/test.yml#L17-L25</a></p>
<pre><code>curl -L https://codeclimate.com/downloads/test-reporter/test-reporter-latest-linux-amd64 &gt; ./cc-test-reporter
chmod +x ./cc-test-reporter
./cc-test-reporter format-coverage -t lcov coverage/lcov.info
./cc-test-reporter upload-coverage
</code></pre>
<p>This works - but seems to be missing a “GIT_BRANCH” variable:</p>
<p><a href="https://docs.codeclimate.com/docs/test-coverage-troubleshooting-branch-names#section-check-your-cis-environmental-variable-names" rel="nofollow noopener">https://docs.codeclimate.com/docs/test-coverage-troubleshooting-branch-names#section-check-your-cis-environmental-variable-names</a></p>
<p>Does anyone have experience with configuring this?</p>

**Comment by AndreaGriffiths11** [comment]
<p>Hi <a class="mention" href="/ksylvest">@ksylvest</a>.,</p>
<p>Thanks for being here and apologies for the delay, we’ve looked internally to see if there were any setup examples with code climate and have not found any workflows to share, I wonder if reaching out to them directly might help? <a href="https://github.com/codeclimate" rel="nofollow noopener">https://github.com/codeclimate</a></p>

**Comment by ethomson** [comment, marked as answer]
<p>I’m afraid that we don’t set the <code>GIT_BRANCH</code> environment variable.  We do, however, set the <code>GITHUB_REF</code> variable, which should be able to be used.  You should be able to remove the leading <code>refs/heads/</code> prefix from that variable and set it to GIT_BRANCH.  For example:</p>
<pre><code>export GIT_BRANCH="${GITHUB_REF/refs\/heads\//}"

curl -L https://codeclimate.com/downloads/test-reporter/test-reporter-latest-linux-amd64 &gt; ./cc-test-reporter
chmod +x ./cc-test-reporter
./cc-test-reporter format-coverage -t lcov coverage/lcov.info
./cc-test-reporter upload-coverage
</code></pre>

**Comment by sigfrid** [comment]
<p>Hi there,<br>
I’m having a hard time making Code Coverage work: I have a GitHub action running CI (<code>RSpec</code> specifications) and once done I post the <code>simplecov</code> outcome to Code Climate. The upload to Code Climate is successful and no error is displayed. However, even if I set the <code>GIT_BRANCH</code> variable, the value passed to Code Climate is always <code>HEAD</code> (I have also unsuccessfully tried to hardcode a test value as GIT_BRANCH).</p>
<p>My steps look like</p>
<pre><code class="lang-auto">- name: Install Code Climate reporter
        run: |
          export GIT_BRANCH="${GITHUB_REF/refs\/heads\//}"
          sudo curl -L https://codeclimate.com/downloads/test-reporter/test-reporter-latest-linux-amd64 &gt; ./cc-test-reporter
          sudo chmod +x ./cc-test-reporter

...
RUN RSPEC
...

- name: Upload coverage information
        env:
          CC_TEST_REPORTER_ID: ${{ secrets.CODE_CLIMATE_ID }}
        run: |
          sudo ./cc-test-reporter after-build -r $CC_TEST_REPORTER_ID
</code></pre>
<p>Any ideas why the <code>GIT_BRANCH</code> isn’t correctly passed to Code Climate?</p>
<p>Thanks.</p>
